build.assets/tooling/cmd/build-os-package-repos/gpg.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"strings"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type GPG struct{}
+
+// Instantiates GPG, ensuring all system requirements for using GPG are fulfilled
+func NewGPG() (*GPG, error) {
+	g := &GPG{}
+
+	err := g.ensureFirstRunHasOccurred()
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to setup GPG")
+	}
+
+	err = g.ensureSecretKeyExists()
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to ensure a secret key exists")
+	}
+
+	return g, nil
+}
+
+// The first time GPG is run for a user with any "meaningful" arguments it will
```

Comment:
Do you need to use the system PGP? If you're just signing things you might be able to get away with `github.com/ProtonMail/go-crypto/openpgp`. See `teleport-plugins/tooling`

Reply:
Did a bunch of reading/testing and neither `golang.org/x/crypto/openpgp` nor `github.com/ProtonMail/go-crypto/openpgp` support keybox (.kbx) files, which the signing keypair use. There are a number of workarounds here, but the only one that doesn't have fairly large downsides is using the system's GPG binary. Several other projects (sops, aptly) use this approach as well.